hikari/impl/stateful_cache.py

```diff
@@ -1598,3 +1605,32 @@ def update_voice_state(
         cached_voice_state = self.get_voice_state(voice_state.guild_id, voice_state.user_id)
         self.set_voice_state(voice_state)
         return cached_voice_state, self.get_voice_state(voice_state.guild_id, voice_state.user_id)
+
+    def clear_messages(self) -> cache.CacheView[snowflakes.Snowflake, messages.PartialMessage]:
+        if not self._message_entries:
+            return cache_utility.EmptyCacheView()
+
+        messages = self._message_entries
+        self._message_entries = collections.LimitedCapacityCacheMap(limit=self.max_messages)
+
+        return cache_utility.StatefulCacheMappingView(messages)
+
+    def delete_message(self, message_id: snowflakes.Snowflake) -> typing.Optional[messages.PartialMessage]:
+        return self._message_entries.pop(message_id) if message_id in self._message_entries else None
+
+    def get_message(self, message_id: snowflakes.Snowflake) -> typing.Optional[messages.PartialMessage]:
+        return self._message_entries.get(message_id)
```

Comment:
you should be copying the message and user object attached to it here (copy.copy, not deepcopy)